### PR TITLE
BAVL-534 this changes adds the availability check to the core API end points, create, amend and request a booking.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
This change replaces what was temporarily removed, a check of clashing prison appointment times (in BVLS only).

**Note this does go further and checks external clashes as well. Do we want to switch the external check off in these cases?**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/RequestBookingService.kt [hunk 1]
```diff
@@ -43,12 +43,21 @@ class RequestBookingService(
...
   private val notificationRepository: NotificationRepository,
   private val locationsInsidePrisonClient: LocationsInsidePrisonClient,
+  private val availabilityService: AvailabilityService,
 ) {
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
-  fun request(request: RequestVideoBookingRequest, user: User) {
+  fun request(request: RequestVideoBookingRequest, user: ExternalUser) {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/RequestBookingService.kt:52]: Only external users should be using this service.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AmendVideoBookingServiceTest.kt [hunk 2]
```diff
@@ -455,42 +454,6 @@ class AmendVideoBookingServiceTest {
     assertThrows<CaseloadAccessException> { service.amend(1, mock<AmendVideoBookingRequest>(), PRISON_USER_RISLEY) }
   }
 
-  @Disabled("Temporary disabling as part of availability ticket BAVL-534. Will be moving clashing check into the facade")
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AmendVideoBookingServiceTest.kt:458]: Removed.  This check is now carried out in the facade.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CreateVideoBookingServiceTest.kt [hunk 3]
```diff
@@ -480,45 +479,6 @@ class CreateVideoBookingServiceTest {
     error.message isEqualTo "Court bookings can only have one pre hearing, one hearing and one post hearing."
   }
 
-  @Disabled("Temporary disabling as part of availability ticket BAVL-534. Will be moving clashing check into the facade")
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CreateVideoBookingServiceTest.kt:483]: Removed.  This check is now carried out in the facade.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/VideoLinkBookingController.kt [hunk 4]
```diff
@@ -205,7 +205,7 @@ class VideoLinkBookingController(
     @Parameter(description = "The request with the requested video link booking details", required = true)
     request: RequestVideoBookingRequest,
     httpRequest: HttpServletRequest,
-  ) = requestBookingService.request(request, httpRequest.getBvlsRequestContext().user)
+  ) = requestBookingService.request(request, httpRequest.getBvlsRequestContext().user as ExternalUser)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/VideoLinkBookingController.kt:208]: Should only be external users using this endpoint.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/RequestBookingService.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/RequestBookingService.kt:52]: There was no availability check in the request a booking journey before? Does it need one now? I think this just sends an email to the prison to book and confirm it. It wouldn't have failed the request for this reason before.

Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/RequestBookingService.kt:52]: I checked with Steve on this one.  There is an availability check done in the UI for request a booking.

**Review by timharrison-moj — APPROVED** [review]
LGTM. Very next. One question on the check in request a booking.

**Pull request merged**